corkscrew.c

```diff
@@ -222,15 +243,25 @@ char *argv[];
 	strncat(uri, destport, sizeof(uri) - strlen(uri) - 1);
 	strncat(uri, " HTTP/1.0", sizeof(uri) - strlen(uri) - 1);
 	if (up != NULL) {
-		strncat(uri, "\nProxy-Authorization: Basic ", sizeof(uri) - strlen(uri) - 1);
-		strncat(uri, base64_encode(up), sizeof(uri) - strlen(uri) - 1);
+		size_t encoded_len = 0;
+		int encode_result = base64_encode(&b64_buf, up, &b64_buf_len, &encoded_len);
+		if (0 == encode_result && encoded_len > 0) {
+			strncat(uri, "\nProxy-Authorization: Basic ", sizeof(uri) - strlen(uri) - 1);
+			strncat(uri, b64_buf, encoded_len);
+		}
+		else {
+			exit_code = 1;
+			fprintf(stderr, "ERROR: base64 encoding failed\n");
```

Comment:
Please word the error message in a similar style as other messages (i.e. avoid the capitalized `ERROR:` prefix).